coreneuron/io/reports/report_handler.cpp

```diff
@@ -155,6 +161,76 @@ VarsToReport ReportHandler::get_compartment_vars_to_report(const NrnThread& nt,
     return vars_to_report;
 }
 
+std::string getSectionTypeStr(SectionType type) {
+    switch (type) {
+        case Axon:
+            return "axon";
+        case Dendrite:
+            return "dend";
+        case Apical:
+            return "apic";
+        default:
+            std::cerr << "SectionType not handled in getSectionTypeStr" << std::endl;
+            nrn_abort(1);
+    }
+}
+
+VarsToReport ReportHandler::get_section_vars_to_report(const NrnThread& nt,
+                                                       const std::set<int>& target,
+                                                       double* report_variable,
+                                                       SectionType section_type,
+                                                       bool all_compartments) const {
+    VarsToReport vars_to_report;
+    const auto& section_type_str = getSectionTypeStr(section_type);
+    const auto* mapinfo = static_cast<NrnThreadMappingInfo*>(nt.mapping);
+    if (!mapinfo) {
+        std::cerr << "[COMPARTMENTS] Error : mapping information is missing for a Cell group "
+                  << nt.ncell << '\n';
+        nrn_abort(1);
+    }
+
+    for (int i = 0; i < nt.ncell; i++) {
+        int gid = nt.presyns[i].gid_;
+        if (target.find(gid) != target.end()) {
+            CellMapping* cell_mapping = mapinfo->get_cell_mapping(gid);
+            if (cell_mapping == nullptr) {
+                std::cerr
+                    << "[COMPARTMENTS] Error : Compartment mapping information is missing for gid "
+                    << gid << '\n';
+                nrn_abort(1);
+            }
+            std::vector<VarWithMapping> to_report;
+            to_report.reserve(cell_mapping->size());
+            
+            /** get section list mapping for the type, if available */
+            if (cell_mapping->get_seclist_section_count(section_type_str) > 0) {
+                SecMapping* s = cell_mapping->get_seclist_mapping(section_type_str);
+                for (const auto& sm : s->secmap) {
+                    int compartment_id = sm.first;
+                    const auto& vec = sm.second;
+
+                    /** get all compartment values (otherwise, just middle point) */
+                    if (all_compartments) {
+                        for (const auto& idx : vec) {
+                            /** corresponding voltage in coreneuron voltage array */
+                            double* variable = report_variable + idx;
+                            to_report.push_back(VarWithMapping(compartment_id, variable));
+                        }
+                    } else {
+                        assert(vec.size() % 2);
```

Comment:
I was thinking after our last discussion, maybe here we should have:
 ```suggestion
                        nrn_assert(vec.size() % 2);
```